xmtp_proto/src/api_client.rs

```diff
@@ -132,7 +132,7 @@ pub type WelcomeMessageStream = Pin<Box<dyn Stream<Item = Result<WelcomeMessage,
 // Wasm futures don't have `Send` or `Sync` bounds.
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
-pub trait XmtpMlsClient: Send + Sync {
```

Comment:
Why I think this is needed:
1. A struct which holds references generally has a generic lifetime parameter attached to it. Downstream users of the struct then have to handle the lifetime, seeing as the reference may expire while the struct is still being held. A struct which holds no references is considered `'static`.
2. Likewise, any downstream users of any trait will have to deal with lifetimes, because you cannot know if the struct that implements the trait holds any references.
3. To handle subscriptions, we need to be able to pass an `Arc<Client>` across threads. Anything passed across to `thread::spawn` must have a `'static` lifetime because it cannot be determined how long the thread will live for.
4. `Client` holds a member that implements the `XmtpMlsClient` trait. Therefore, `Client` is only `'static` if the implementer of `XmtpMlsClient` is.

Here, we specify that the implementer of `XmtpMlsClient` is always `'static` - that is, it is a struct that either holds no references, or only holds references with `'static` lifetime.

It's possible to only specify `'static` bounds on `XmtpMlsClient` wherever streaming is used, and leave open the option of having an `XmtpMlsClient` which is not `'static` and forgo streaming. I chose this way because it is less duplicative elsewhere and I don't see any need for a non-static `XmtpMlsClient`.

Reply:
Static lifetime doesn't seem that bad here, given you only have one `XmtpMlsClient` for each client instance. Even in the bindings we keep around the client for a long time. It's not like you are going to have thousands of these floating around over time, except maybe in some weird bot edge cases.

Reply:
To be clear, I don't think it's about how long `XmtpMlsClient` lives for, because it is a struct and not a reference. What I'm saying is all structs are automatically `'static` unless they hold a reference that is not `'static`. What I believe I'm doing here is enforcing that whatever implements `XmtpMlsClient` doesn't hold non-static references